src/utils/EventEmitter.ts

```diff
@@ -8,24 +8,30 @@ function uuidv4(): string {
 
 /** Generic Event Emitter class following a subscribe/publish paradigm. */
 export class EventEmitter<T extends any[]> {
-  subscribers: Record<string, (...args: [...T]) => void | undefined> = {};
+  private subscribers: Map<string, (...args: [...T]) => void | undefined> = new Map();
```

Comment:
Optional: I don't think we use typescript's access-control declarations (private, protected) elsewhere in the codebase. At minimum, they're very rare. I can certainly see the value if they were used consistently, but if they're only used in a couple of places they could get confused with JS's private variables. 
Anyway, up to you.